docs/sources/operations/storage/retention.md

```diff
@@ -3,7 +3,157 @@ title: Retention
+      - from: "2020-07-31"
+        index:
+            period: 24h
+            prefix: loki_index_
+        object_store: gcs
+        schema: v11
+        store: boltdb-shipper
+storage_config:
+    boltdb_shipper:
+        active_index_directory: /data/index
+        cache_location: /data/boltdb-cache
+        shared_store: gcs
+    gcs:
+        bucket_name: loki
+```
+
+> Note that retention is only available if the index period is 24h.
+
+As you can see the retention needs `schema_config` and `storage_config` to access the storage.
+To activate retention you'll need to set `retention_enabled` to true otherwise the compactor will only compact tables.
+
+The `working_directory` is the directory where marked chunks and temporary tables will be saved.
+The `compaction_interval` dictates how often retention and/or compaction is applied, if it falls behind it will execute as soon as possible.
+`retention_delete_delay` is the delay after which the compactor will actually delete marked chunks.
+
+And finally `retention_delete_worker_count` setup the amount of goroutine to use to delete chunks.
+
+> Compaction and retention is idempotent, in case of restart the compactor will restart where he left off.
+
+#### Configuring retention period
+
+So far we've only talked about how to setup the compactor to apply retention, in this section we will explain how to configure the actual retention period to apply.
+
+Retention period is configure via the [`limits_config`](./../../../configuration/#limits_config) configuration section. It supports two type of retention `retention_stream` which is applied to all chunks matching the selector
+
+The example below can be added to your Loki configuration to configure global retention.
+
+```yaml
+...
+limits_config:
+  retention_period: 744h
+  retention_stream:
+  - selector: '{namespace="dev"}'
+    priority: 1
+    period: 24h
+  per_tenant_override_config: /etc/overrides.yaml
+...
+```
+
+Per tenant retention can be defined using the `/etc/overrides.yaml` files. For example:
+
+```yaml
+overrides:
+    "29":
+        retention_period: 168h
+        retention_stream:
+        - selector: '{namespace="prod"}'
+          priority: 2
+          period: 336h
+        - selector: '{container="loki"}'
+          priority: 1
+          period: 72h
+    "30":
+        retention_stream:
+        - selector: '{container="nginx"}'
+          priority: 1
+          period: 24h
+```
+
+The most specific matching rule is selected. The rule selection is as follow:
+
+- If a per-tenant `retention_stream` matches the current stream, the highest priority is picked.
+- Otherwise if a global `retention_stream` matches the current stream, the highest priority is picked.
+- Otherwise if a per-tenant `retention_period` is specified it will be applied.
+- The the global `retention_period` will be selected if nothing else matched.
+- And finally if no global `retention_period` is set, the default `30days (744h)` retention is selected.
```

Comment:
Revise lines 127 thru 133 with::

A rule to apply is selected by choosing the first in this list that matches:
1. If a per-tenant `retention_stream` matches the current stream, the highest priority is picked.
1.  If a global `retention_stream` matches the current stream, the highest priority is picked.
1. If a per-tenant `retention_period` is specified, it will be applied.
1. The global `retention_period` will be selected if nothing else matched.
1. If no global `retention_period` is specified, the default value of `744h` (30days) retention is used.